acceptancetests/jujupy/workloads.py

```diff
@@ -0,0 +1,141 @@
+    [wiki_unit_name] = [
+        k for k, v in status.get_applications()['mediawiki']['units'].items()
+        if v.get('leader', False)]
+    wiki_ip = get_unit_ipaddress(client, wiki_unit_name)
+    resp = requests.get('http://{}'.format(wiki_ip))
+    if not resp.ok:
+        raise AssertionError('Mediawiki not responding; {}: {}'.format(
+            resp.status_code, resp.reason
+        ))
+    if '<title>Please set name of wiki</title>' not in resp.content:
+        raise AssertionError('Got unexpected mediawiki page content.')
+
+
+def deploy_simple_server_to_new_model(
+        client, model_name, resource_contents=None, series='xenial'):
+    # As per bug LP:1709773 deploy 2 primary apps and have a subordinate
+    #  related to both
+    new_model = client.add_model(client.env.clone(model_name))
+    application = deploy_simple_resource_server(
+        new_model, resource_contents, series)
+    _, deploy_complete = new_model.deploy('cs:ubuntu', series=series)
+    new_model.wait_for(deploy_complete)
+    new_model.deploy('cs:nrpe', series=series)
+    new_model.juju('add-relation', ('nrpe', application))
+    new_model.juju('add-relation', ('nrpe', 'ubuntu'))
+    # Need to wait for the subordinate charms too.
+    new_model.wait_for(AllApplicationActive())
+    new_model.wait_for(AllApplicationWorkloads())
+    new_model.wait_for(AgentsIdle(['nrpe/0', 'nrpe/1']))
+    assert_deployed_charm_is_responding(new_model, resource_contents)
+
+    return new_model, application
+
+
+def deploy_simple_resource_server(
+        client, resource_contents=None, series='xenial'):
+    application_name = 'simple-resource-http'
+    log.info('Deploying charm: '.format(application_name))
+    charm_path = local_charm_path(
+        charm=application_name, juju_ver=client.version)
+    # Create a temp file which we'll use as the resource.
+    if resource_contents is not None:
+        with temp_dir() as temp:
+            index_file = os.path.join(temp, 'index.html')
+            with open(index_file, 'wt') as f:
+                f.write(resource_contents)
+            client.deploy(
+                charm_path,
+                series=series,
+                resource='index={}'.format(index_file))
+    else:
+        client.deploy(charm_path, series=series)
+
+    client.wait_for_started()
+    client.wait_for_workloads()
+    client.juju('expose', (application_name))
+    return application_name
+
+
+def deploy_dummy_source_to_new_model(client, model_name):
+    new_model_client = client.add_model(client.env.clone(model_name))
+    charm_path = local_charm_path(
+        charm='dummy-source', juju_ver=new_model_client.version)
+    new_model_client.deploy(charm_path)
+    new_model_client.wait_for_started()
+    new_model_client.set_config('dummy-source', {'token': 'one'})
+    new_model_client.wait_for_workloads()
+    return new_model_client
+
+
+def assert_deployed_charm_is_responding(client, expected_output=None):
```

Comment:
this method name is really generic. If it's not a generic method, let's be explicit about what it's doing.